### DISCUSSION TITLE
Where should proposals for changes to Tenacity be discussed?

### DISCUSSION THREAD

**Discussion by n0toose** [tenacityteam/tenacity-legacy#651] [Q&A] [answered]
Hi, we've recently noticed a huge problem in our development process that ultimately hampers our productivity and our ability to make a first release. It's also a thing that generally exists in free and open source software development and constitutes it as very hostile towards beginners that are just starting out.

Proposed changes to our code. Also known as "pull requests" on GitHub, or "patches" elsewhere. There are many times where a contributor may devote hours, days, or even weeks to a specific change, only to have these shot down by another contributor because of a fundamental disagreement with the change that should have been probably brought up earlier before all of that time was devoted.

We want to figure out a way that we'll be able to debate and agree on preliminary changes (e.g. changing the order of some icons), and then only evaluate the implementation that was used for that said change (did it work?) afterwards. It's important that the developers/contributors will also remain engaged later on, just so that we won't have any unpleasant surprises.

What do you think? (Please consider that recommendations made by existing contributors/maintainers may have more gravity. Nevertheless, we strive to be community-led, which also means allowing the community to help us, so we are open to all feedback.)

---

Summarized list of ideas (I'll keep this updated):
- Mailing List
- GitHub Discussions

**Comment by nbsp** [comment, marked as answer]
I don't think there's much of a difference between mailing lists and GitHub Discussions, apart from Markdown, and the need for a GitHub account. I think the mailing list is more appropriate.

**Reply by n0toose** [reply]
Cool, can we create one?

**Reply by leio** [reply]
What's wrong with one of the existing ones for this like tenacity-dev?

Personally I prefer IRC discussions and maybe discussions here, not yet another mailing list (to my dozens from other projects) with less real-time communication, but that's that.
tenacity-dev doesn't seem to be crowded with anything, so at least I don't see a reason to make another one.

**Reply by nbsp** [reply]
reminder that [tenacity-discuss](https://lists.sr.ht/~tenacity/tenacity-discuss) also exists, but I think dev discussion should be kept to the development mailing list or IRC.

**Reply by n0toose** [reply]
TO-DO: Document [development mailing list](https://lists.sr.ht/~tenacity/tenacity-dev) as the place to share ideas before starting to work on them in a very obvious place.

**Reply by JPcode05** [reply]
Have the community forums on Discourse (forums.tenacityaudio.org), and #tenacity on IRC (and optionally the tenacity-discuss mailing list), and the development mailing list tenacity-dev and #tenacity-dev on IRC.

Something I would like to know about is that the mailing lists on Sourcehut are plaintext only, and we need accomodation for users of Gmail, Outlook, Webmail, etc. where formatting might get complicated. Will those make it through to Sourcehut? By that, I mean, how do we deal with those who forgot to send mail in plaintext?

**Reply by nbsp** [reply]
> Something I would like to know about is that the mailing lists on Sourcehut are plaintext only, and we need accomodation for users of Gmail, Outlook, Webmail, etc. where formatting might get complicated. Will those make it through to Sourcehut? By that, I mean, how do we deal with those who forgot to send mail in plaintext?

Plaintext email isn't handled by the server. Gmail users can use a client which respects plaintext email, and not run into any problems. Regardless, sourcehut notifies the sender if their email was rejected, so the email wouldn't silently drop.
